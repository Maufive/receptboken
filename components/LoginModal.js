import React, { Component } from "react";
import Modal from "react-modal";
import { MessageConsumer } from "./providers/MessageProvider";
import Login from "./Login";
import { Button } from "./styles/Button";
import { LoginContainer, customStyles } from "./styles/Jumbotron";

Modal.setAppElement("#__next");

class LoginModal extends Component {
	state = {
		isModalOpen: false
	};

	componentDidMount = () => {
		this.setState({ isModalOpen: this.props.open });
	};

	openModal = () => {
		this.setState({ isModalOpen: true });
	};

	closeModal = () => {
		this.setState({ isModalOpen: false });
	};

	render() {
		return (
			<LoginContainer>
				{this.props.primaryButton ? (
					<Button fullWidth primary onClick={this.openModal}>
						Logga in
					</Button>
				) : (
					<Button fullWidth onClick={this.openModal}>
						Logga in
					</Button>
				)}
				<Modal
					isOpen={this.state.isModalOpen}
					onAfterOpen={this.afterOpenModal}
					onRequestClose={this.closeModal}
					style={customStyles}
					contentLabel="Login Modal"
				>
					<MessageConsumer>
						{({ setMessage }) => (
							<Login setUser={this.props.setUser} setMessage={setMessage} />
						)}
					</MessageConsumer>
				</Modal>
			</LoginContainer>
		);
	}
}

export default LoginModal;
