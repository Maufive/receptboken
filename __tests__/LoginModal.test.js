import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import LoginModal from "../components/LoginModal";

describe("<LoginModal />", () => {
	let wrapper;
	beforeEach(() => (wrapper = shallow(<LoginModal />)));

	it("renders with closed state", () => {
		expect(wrapper.find("Button__MinButton").text()).toBe(
			"<LockIcon />Logga in"
		);
		expect(wrapper.find("Modal").prop("isOpen")).toEqual(false);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});

	it("renders with open state", () => {
		wrapper.setState({ isModalOpen: true });
		expect(wrapper.find("Modal").prop("isOpen")).toEqual(true);
	});

	it("opens modal when button is clicked", () => {
		wrapper.find("Button__MinButton").simulate("click");
		wrapper.update();
		expect(wrapper.state("isModalOpen")).toBe(true);
	});
});
