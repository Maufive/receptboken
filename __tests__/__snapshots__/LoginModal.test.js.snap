// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginModal /> renders with closed state 1`] = `
<Jumbotron__LoginContainer>
  <Button__MinButton
    onClick={[Function]}
  >
    <LockIcon
      viewBox="0 0 54 54"
      xmlns="http://www.w3.org/2000/svg"
    />
    Logga in
  </Button__MinButton>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    closeTimeoutMS={0}
    contentLabel="Login Modal"
    isOpen={false}
    onRequestClose={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {
          "background": "#fff",
          "bottom": "auto",
          "left": "50%",
          "marginRight": "-50%",
          "right": "auto",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "zIndex": 10,
        },
      }
    }
  >
    <ContextConsumer>
      <Component />
    </ContextConsumer>
  </Modal>
</Jumbotron__LoginContainer>
`;
